pkg/chrootarchive/archive_windows.go

```diff
@@ -12,7 +12,7 @@ func chroot(path string) error {
 	return nil
 }
 
-func invokeUnpack(decompressedArchive io.ReadCloser,
+func invokeUnpack(decompressedArchive io.Reader,
```

Comment:
because https://github.com/docker/docker/pull/16932/files#diff-7d8fc0dcb0916621d523276add8210c2R63 you call it with `r` which is now an `io.Reader` not an `io.ReadCloser` (since you removed the nopcloser wrapper). And we build binaries for all platforms, not just linux.